File: ROS_Exercises/2-Exercise.2/Python/exercise2_number_publisher.py
#!/usr/bin/env python

import rospy
from std_msgs.msg import Int64
from random import randint

if __name__ == '__main__':
	rospy.init_node("number_publisher")
	pub = rospy.Publisher("/number", Int64, queue_size=20)

	rospy.set_param("/number_publish_frequency", 1)
	publish_frequency = rospy.get_param("/number_publish_frequency")
	rate = rospy.Rate(publish_frequency)
	
	while not rospy.is_shutdown():
		msg = Int64()
		rand = randint(0, 9)
		msg.data = rand
		pub.publish(msg)
		rate.sleep()
	rospy.loginfo("Node was stopped!!!")

File: ROS_Exercises/2-Exercise.2/C++/exercise2_number_publisher.cpp
#include <ros/ros.h>
#include <std_msgs/Int64.h>

int main (int argc, char **argv)
{
	ros::init(argc, argv, "number_publisher", ros::init_options::AnonymousName);
	ros::NodeHandle nh;
	ros::Publisher pub = nh.advertise<std_msgs::Int64>("/number", 10);

	nh.setParam("/number_publish_frequency", 1);
	double publish_frequency;
	nh.getParam("/number_publish_frequency", publish_frequency);
	ros::Rate rate(publish_frequency);

	while (ros::ok()) {
		std_msgs::Int64 msg;
		int iRand = rand() % 10;
		msg.data = iRand;
		pub.publish(msg);
		rate.sleep();
	}
}
